Sources/Tuvi.Core.DataStorage.Impl/PasswordProtectedStorage.cs

```diff
@@ -207,6 +208,30 @@ private static async Task MigrateAsync(SQLiteAsyncConnection db)
 
             // Automatic migration
             await CreateOrMigrateTablesAsync(db).ConfigureAwait(false);
+
+            // --- Data migration (18.05.2025): Account.EmailId -> Account.Email ---
+            // Get all accounts where EmailAddress is not set, but EmailId is present
+            var accounts = await db.Table<Account>().ToListAsync().ConfigureAwait(false);
+            bool needMigration = accounts.Any(a => string.IsNullOrEmpty(a.EmailAddress) && a.EmailId != 0);
+
+            if (needMigration)
+            {
+                foreach (var account in accounts)
+                {
+                    // If EmailAddress is not set and EmailId is present, migrate the data
+                    if (string.IsNullOrEmpty(account.EmailAddress) && account.EmailId != 0)
+                    {
+                        var emailData = await db.FindAsync<EmailAddressData>(account.EmailId).ConfigureAwait(false);
+                        if (emailData != null)
+                        {
+                            // Set the Email property using the data from EmailAddressData
+                            account.Email = new EmailAddress(emailData.Address, emailData.Name);
+                            account.EmailId = 0; // Reset EmailId to avoid future migrations
```

Comment:
EmailId value is used in other entities:
- [class Contact](https://github.com/Eppie-io/TuviCore/blob/fecc44de47b774501704b609b59f0ef4ef477c01/Sources/Tuvi.Core.Entities/Contact.cs#L51)
- [class LastMessageData](https://github.com/Eppie-io/TuviCore/blob/fecc44de47b774501704b609b59f0ef4ef477c01/Sources/Tuvi.Core.Entities/LastMessageData.cs#L13)

Reply:
Contact.EmailId is not associated with Account and does not need to be changed.
LastMessageData.AccountEmailId changed to AccountId

Reply:
if `Contact.EmailId` is not used, mark it as deprecated or remove it

Reply:
`Contact.EmailId` is used and needed. It works correctly, it doesn't need to be redesigned. Only `Account.EmailId` should have been fixed.